util/vec-map/Cargo.toml

```diff
@@ -8,4 +8,4 @@ readme = "README.md"
 
 [dependencies]
 displaydoc = "0.2"
-heapless = { version = "0.8", default-features = false }
+heapless = { git = "https://github.com/japaric/heapless", branch = "main", default-features = false }
```

Comment:
Oh, why is this needed?
At the very least we should point at a specific commit and not a branch.

Reply:
0.8.0 isn't available in the crates.io, ok Ill update to a commit hash

Reply:
What was the reason this needed to be updated in the first place?

Reply:
Couple other points:
1. It shouldn't track main, which presumably will change over time (i.e. it should track the diff that introduces this)
2. The recommended way to do this is to patch the crates.io registry to use a git commit at a specific revision via the workspace `Cargo.toml` files (e.g. at the top-level and `<thing>/enclave/trusted/Cargo.toml`)

Reply:
> hat was the reason this needed to be updated in the first place?

ya, Ive been trying to repro the "dependency issue" that made me update `heapless` but have been unsuccessful. Ill try a few more but maybe we don't need it.

Reply:
can't repro the dependency issue that I originally had that motivated these other changes, so I've removed them and force pushed just the mbedtls commit hash change

Reply:
This is the dependency error if I only update mbedTLS.

```console
error: failed to select a version for `spin`.
    ... required by package `mbedtls v0.8.1 (https://github.com/mobilecoinfoundation/rust-mbedtls.git?rev=6f5bd68166a2c12fa3922867a6a6bd68904243d3#6f5bd681)`
    ... which satisfies dependency `mbedtls = "^0.8.1"` of package `mc-attest-net v4.0.0 (/__w/mobilecoin/mobilecoin/attest/net)`
    ... which satisfies path dependency `mc-attest-net` (locked to 4.0.0) of package `mc-attest-ake v4.0.0 (/__w/mobilecoin/mobilecoin/attest/ake)`
    ... which satisfies path dependency `mc-attest-ake` (locked to 4.0.0) of package `mc-attest-api v4.0.0 (/__w/mobilecoin/mobilecoin/attest/api)`
    ... which satisfies path dependency `mc-attest-api` (locked to 4.0.0) of package `mc-connection v4.0.0 (/__w/mobilecoin/mobilecoin/connection)`
    ... which satisfies path dependency `mc-connection` (locked to 4.0.0) of package `mc-connection-test-utils v4.0.0 (/__w/mobilecoin/mobilecoin/connection/test-utils)`
    ... which satisfies path dependency `mc-connection-test-utils` (locked to 4.0.0) of package `mc-ledger-sync v4.0.0 (/__w/mobilecoin/mobilecoin/ledger/sync)`
    ... which satisfies path dependency `mc-ledger-sync` (locked to 4.0.0) of package `mc-consensus-service v4.0.0 (/__w/mobilecoin/mobilecoin/consensus/service)`
versions that meet the requirements `^0.9.4` are: 0.9.4

all possible versions conflict with previously selected packages.

  previously selected package `spin v0.9.3`
    ... which satisfies dependency `spin = "^0.9.2"` (locked to 0.9.3) of package `heapless v0.7.16`
    ... which satisfies dependency `heapless = "^0.7"` (locked to 0.7.16) of package `mc-util-vec-map v4.0.0 (/__w/mobilecoin/mobilecoin/util/vec-map)`
    ... which satisfies path dependency `mc-util-vec-map` (locked to 4.0.0) of package `mc-transaction-extra v4.0.0 (/__w/mobilecoin/mobilecoin/transaction/extra)`
    ... which satisfies path dependency `mc-transaction-extra` (locked to 4.0.0) of package `mc-api v4.0.0 (/__w/mobilecoin/mobilecoin/api)`
    ... which satisfies path dependency `mc-api` (locked to 4.0.0) of package `mc-consensus-api v4.0.0 (/__w/mobilecoin/mobilecoin/consensus/api)`
    ... which satisfies path dependency `mc-consensus-api` (locked to 4.0.0) of package `mc-connection v4.0.0 (/__w/mobilecoin/mobilecoin/connection)`
    ... which satisfies path dependency `mc-connection` (locked to 4.0.0) of package `mc-connection-test-utils v4.0.0 (/__w/mobilecoin/mobilecoin/connection/test-utils)`
    ... which satisfies path dependency `mc-connection-test-utils` (locked to 4.0.0) of package `mc-ledger-sync v4.0.0 (/__w/mobilecoin/mobilecoin/ledger/sync)`
    ... which satisfies path dependency `mc-ledger-sync` (locked to 4.0.0) of package `mc-consensus-service v4.0.0 (/__w/mobilecoin/mobilecoin/consensus/service)`
```

the full fix now depends [on a fork of `heapless`](https://github.com/mobilecoinofficial/heapless/tree/adam/update-spin) where I increment the version of `spin` to `0.9.4`.